runtime/vm/JFRConstantPoolTypes.hpp

```diff
@@ -1246,6 +1274,64 @@ class VM_JFRConstantPoolTypes {
 		gcConfiguration->heapAddressBits = J9JAVAVM_REFERENCE_SIZE(vm) * 8;
 	}
 
+	static uintptr_t processNativeLibrariesCallback(const char *libraryName, void *lowAddress, void *highAddress, void *userData)
+	{
+		size_t libraryPathLength = strlen(libraryName);
+		VM_JFRConstantPoolTypes *constantPoolTypes = reinterpret_cast<VM_JFRConstantPoolTypes *>(userData);
+		J9Pool *nativeLibrariesTable = constantPoolTypes->_nativeLibrariesTable;
+		NativeLibraryEntry *firstNativeLibraryEntry = constantPoolTypes->_firstNativeLibraryEntry;
+		NativeLibraryEntry *previousNativeLibraryEntry = constantPoolTypes->_previousNativeLibraryEntry;
+		NativeLibraryEntry *entry = firstNativeLibraryEntry;
+		PORT_ACCESS_FROM_JAVAVM(constantPoolTypes->_vm);
+		for (; NULL != entry; entry = entry->next) {
+			if (0 == strcmp(entry->name, libraryName)) {
+				if (entry->addressLow > (UDATA)lowAddress) {
+					entry->addressLow = (UDATA)lowAddress;
+				}
+				if (entry->addressHigh < (UDATA)highAddress) {
+					entry->addressHigh = (UDATA)highAddress;
+				}
+				return 0;
+			}
+		}
+		char *libraryNameCopy = (char *)j9mem_allocate_memory(libraryPathLength + 1, OMRMEM_CATEGORY_VM);
+		if (NULL == libraryNameCopy) {
+			/* Allocation for library name failed. */
+			return 1;
+		}
+		memcpy(libraryNameCopy, libraryName, libraryPathLength + 1);
```

Comment:
Please move this call to `memcpy()` after line 1308 (here it would be wasted work if a new pool entry cannot be allocated).